### PR TITLE
build: simplify, add extras_require, tf variants

### PR DESCRIPTION
**Pull request by brianmartin**
This commit makes several changes to our packaging and build config:

- use extras_require rather than multiple `reqs-*.txt`
- split tox tasks into more granular pieces
- simplify `.circleci/config` using new syntax
  - less verbose
  - removes caching, which wasn't helping wall-clock performance
- test against multiple older tf versions
  - note: currently not running against >1.10.1 due to #83, but I've retained in the `tox.ini`
- remove cruft from `setup.cfg`

----

I don't see this is as the final list of extras, but it keeps parity (i.e. if we were to publish right now) and we can finalize in a follow-up commit.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,69 +1,62 @@
-version: 2
-jobs:
-  test-python3:
+version: 2.1
+
+executors:
+  tox:
     docker:
-      - image: circleci/python:3.6.7
```
Comment (andrewsmartin) [inline .circleci/config.yml]: Do you think we could run into any issues with CircleCI in not using their pre-baked images? (Not that I can think of any per se, but worth making sure)

File: setup.cfg [hunk 2]
```diff
@@ -13,7 +13,6 @@ classifier =
     Operating System :: OS Independent
     Programming Language :: Python
     Programming Language :: Python :: 2.7
-    Programming Language :: Python :: 3
```
Comment (andrewsmartin) [inline setup.cfg]: Why? We'd still like this to work for python 3 right?

File: setup.cfg [hunk 3]
```diff
@@ -22,13 +21,34 @@ keywords =
 packages =
     spotify_tensorflow
 
-[options.extras_require]
-tensorflow = tensorflow==1.9.0
-tensorflow_gpu = tensorflow-gpu==1.9.0
+[extras]
```
Comment (andrewsmartin) [inline setup.cfg]: This is equivalent to `extras_required` in `setup.py` right?

File: setup.cfg [hunk 4]
```diff
@@ -22,13 +21,34 @@ keywords =
...
+tensorflow = 
+    tensorflow==1.9.0
```
Comment (andrewsmartin) [inline setup.cfg]: Should we avoid pinning versions here (leave unpinned or use bounds), and instead add bounds to all deps in `requirements.txt`?

File: setup.cfg [hunk 5]
```diff
@@ -22,13 +21,34 @@ keywords =
...
+tensorflow_gpu = 
+    tensorflow-gpu==1.9.0
+tfx =
+    luigi>=2.3.3,<3.0.0
+    google-api-python-client>=1.5.4,<2.0
+    oauth2client<4.0.0
+    tensorflow-data-validation==0.11.0
```
Comment (andrewsmartin) [inline setup.cfg]: ditto: pinning

File: setup.cfg [hunk 6]
```diff
@@ -22,13 +21,34 @@ keywords =
...
     spotify_tensorflow
 
-[options.extras_require]
-tensorflow = tensorflow==1.9.0
-tensorflow_gpu = tensorflow-gpu==1.9.0
+[extras]
+tensorflow = 
+    tensorflow==1.9.0
+tensorflow_gpu = 
+    tensorflow-gpu==1.9.0
```
Comment (andrewsmartin) [inline setup.cfg]: ditto: pinning

File: setup.cfg (hunk 2 above)
Comment (andrewsmartin) [reply setup.cfg]: nvm - guess we're blocked on beam python.

File: .circleci/config.yml [hunk 7]
```diff
@@ -1,69 +1,62 @@
...
-          name: Upload Coverage
-          command: |
-            pip install --user codecov
-            python -m codecov
+jobs:
 
-  test-python2:
-    docker:
-      - image: circleci/python:2.7.15
+  lint:
```
Comment (andrewsmartin) [inline .circleci/config.yml]: Is it possible to make this more concise by just invoking tox which by default executes all environments? Or is the idea to leverage CircleCI parallelism (are jobs parallelized)?

File: setup.cfg (hunk 4 above)
Comment (brianmartin) [reply setup.cfg]: Currently, `pbr` is implicitly reading the `requirements.txt` and using that to populate `install_requires`.

So until we ditch `pbr`, we can't use the `requirements.txt` in the way we've discussed (i.e. reqs.txt has pinned versions, setup.py has bounds).

Also, this is not really a diff from the master branch. I've just added a line break.

File: setup.cfg (hunk 5 above)
Comment (brianmartin) [reply setup.cfg]: See comment above.

Also, this is not a change from master, where tfdv is pinned in `test-requirements.txt`

File: setup.cfg (hunk 6 above)
Comment (brianmartin) [reply setup.cfg]: ditto

File: .circleci/config.yml (hunk 7 above)
Comment (brianmartin) [reply .circleci/config.yml]: Yes, all the jobs are parallelized. Also, if we used a single tox invocation we'd only get a single "check" on our GH PRs.

File: setup.cfg (hunk 4 above)
Comment (andrewsmartin) [reply setup.cfg]: Oh right - yep, makes sense to wait until we ditch pbr.

File: .circleci/config.yml (hunk 7 above)
Comment (andrewsmartin) [reply .circleci/config.yml]: Cool - very nice.

File: .circleci/config.yml (hunk 1 above)
Comment (andrewsmartin) [reply .circleci/config.yml]: @brianmartin regarding above comment: any thoughts on using the circleci python container instead and adding a `pip install tox` step first?

Comment (brianmartin) [reply .circleci/config.yml]: We certainly could use the circle image. The nice thing about this tox container is that it contains the full set of python versions. Any reason to prefer theirs? I'm curious what your concern is, as using this tox image seems simpler. Is it a trust concern?

**Pull request merged**